src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -523,44 +457,43 @@ private void ProcessProjectChange(Solution solution, ProjectId projectId)
             }
 
             // Project was changed in some way.  Let's go find the set of installed packages for it.
-            var dteProject = _workspace.TryGetDTEProject(projectId);
-            if (dteProject == null)
+            var projectGuid = _workspace.GetProjectGuid(projectId);
+            if (projectGuid == Guid.Empty)
             {
                 // Don't have a DTE project for this project ID.  not something we can query NuGet for.
                 return;
             }
 
-            var installedPackages = new MultiDictionary<string, string>();
-            var isEnabled = false;
+            var serviceContainer = _serviceProvider.GetService<IBrokeredServiceContainer, SVsBrokeredServiceContainer>();
+            var serviceBroker = serviceContainer.GetFullAccessServiceBroker();
+            var nugetService = await serviceBroker.GetProxyAsync<INuGetProjectService>(NuGetServices.NuGetProjectServiceV1).ConfigureAwait(false);
 
-            // Calling into NuGet.  Assume they may fail for any reason.
             try
             {
-                var installedPackageMetadata = _packageInstallerServices.Value.GetInstalledPackages(dteProject);
-                foreach (var metadata in installedPackageMetadata)
+                cancellationToken.ThrowIfCancellationRequested();
+                var installedPackagesResult = await nugetService.GetInstalledPackagesAsync(projectGuid, cancellationToken).ConfigureAwait(false);
+
+                var installedPackages = new MultiDictionary<string, string>();
+                var isEnabled = installedPackagesResult?.Status == InstalledPackageResultStatus.Successful;
+                if (isEnabled)
                 {
-                    if (metadata.VersionString != null)
+                    foreach (var installedPackage in installedPackagesResult.Packages)
                     {
-                        installedPackages.Add(metadata.Id, metadata.VersionString);
+                        var version = installedPackage.RequestedRange ?? installedPackage.Version;
```

Comment:
so "requested range" is what you wrote in your project file.  i.e. i want from 1-2.  'Version' is what is actually resolved to.  we effectively always want 'requested range' so that when we update your other project, it matches your existing projects. *However*, there is an issue currently where Nuget doesn't always return RequestedRange, so we are resilient here to this.  I will doc accordingly.

Reply:
doc'ed this.